Extend the lib with DevlinkPortFnCapSet and DevlinkDevParamSet fns

almaslennikov opened the pull request:
Add support for `port function cap set` and `dev param set` operations.

DevlinkPortFnCapSet sets roce and max_uc_macs port function cap attributes.
Equivalent to: `mlxdevm port function cap sep $port roce true max_uc_macs 64`
Equivalent to: `mlxdevm port function cap sep $port roce false max_uc_macs 128`

DevlinkDevParamSet sets one device parameter. Equivalent to: `mlxdevm dev param set $dev name disable_netdev value true cmode runtime`

almaslennikov committed:
Add DevlinkPortFnCapSet function

DevlinkPortFnCapSet sets roce and max_uc_macs port fn cap attributes
Equivalent to the following mlxdevm command:
`mlxdevm port function cap sep $port roce true max_uc_macs 128`
`mlxdevm port function cap sep $port roce false max_uc_macs 64`

roce parameter allows to enable/disable RDMA over Converged Ethernet
max_uc_macs parameter controls the max length of the unicast MACs list
that the port is able to store

A call example:

```
var fnCapAttr mlxdevm.DevlinkPortFnCapSetAttrs

fnCapAttr.FnCapAttrs.Roce = true
fnCapAttr.RoceValid = true
fnCapAttr.FnCapAttrs.UCList = 64
fnCapAttr.UCListValid = true

socket := "mlxdevm"
bus := "pci"
device := "0000:3b:00.0"
port := uint32(32769)

mlxdevm.DevlinkPortFnCapSet(socket, bus, device, port, fnCapAttr)
```

Signed-off-by: amaslennikov <amaslennikov@nvidia.com>

paravmellanox reviewed (commented):
overall looks fine.
it would have been best if this comes as two patches.

patch 1 to set devlink dev params.
patch 2 to set devlink port function attributes (roce, uc list)

paravmellanox reviewed (commented):
Commit log is bit empty.

An examples along with short details of what functionality (roce, uc list) and devlink param in each patch should be added.

almaslennikov committed:
Add DevlinkDevParamSet function

This function allows to set one device parameter.
Equivalent to the following mlxdevm command:
mlxdevm dev param set $dev name disable_netdev value true cmode runtime

The function first requests the parameter type from the kernel,
then validates the passed new value and send the request to set it.

A call example:

```
socket := "mlxdevm"
bus := "auxiliary"
device := "mlx_core.sf.5"
paramName := "disable_netdev"
newValue := "true"
cmode := "runtime"
mlxdevm.DevlinkDevParamSet(socket, bus, device, paramName, newValue, cmode)
```

Signed-off-by: amaslennikov <amaslennikov@nvidia.com>

almaslennikov commented:
Hi @paravmellanox,

Addressed your comments. Could you take another look?

almaslennikov committed:
Add DevlinkDevParamGet function
This function allows to get one device parameter.
Equivalent to the following mlxdevm command:
mlxdevm dev param show $dev name disable_netdev

The function parses kernel response into a DevlinkDevParam struct
A call example:

```
socket := "mlxdevm"
bus := "auxiliary"
device := "mlx_core.sf.5"
paramName := "disable_netdev"
mlxdevm.DevlinkDevParamGet(socket, bus, device, paramName)
```

Signed-off-by: amaslennikov <amaslennikov@nvidia.com>
